grails-app/services/org/pih/warehouse/forecasting/ForecastingService.groovy

```diff
@@ -27,6 +30,7 @@ import org.pih.warehouse.core.SynonymTypeCode
 
 class ForecastingService {
 
+    HibernateSessionService hibernateSessionService
```

Comment:
I tried for a while to make this use the PersistenceService interface but it wouldn't autowire in HibernateSessionService. According to the docs that I looked at it should have worked because I have the `@Primary` annotation on HibernateSessionService so I'm not sure...

Reply:
That's really weird, we should investigate that. You could check if you use `@Qualifier` if it would autowire then, but still we shouldn't need to do that.....

Reply:
I think I figured it out. If I add the `@Autowired` annotation, it works. Maybe there's some Grails weirdness going on that makes it unable to resolve the bean so we need to rely on spring :shrug: